src/main/java/net/coderbot/iris/mixin/MixinShaderInstance.java

```diff
@@ -55,9 +50,7 @@
 			return;
 		}
 
-		// TODO: Don't hardcode this list of samplers bound to texture unit 0
-		if (samplerName.equals("Sampler0") || samplerName.equals("tex") || samplerName.equals("texture")
-				|| samplerName.equals("gtexture")) {
+		if (samplerUnit == 0) {
```

Comment:
While this addresses a TODO, I'm not much more of a fan of this approach... I think it's worth documenting the fact that this strongly assumes that gtexture always gets bound to texture unit 0, while in the future I'd like to get to the point where this isn't required.

Reply:
Doesn’t it need to be bound to 0, to deal with unrecognized samplers?

Reply:
That's a driver quirk. I'd like to eventually have code that finds the samplers and figures out which ones are unrecognized instead eventually instead of just relying on driver quirks.

Reply:
Sounds good, will changing to TextureUnit.TERRAIN.unitId fix this for now?

Reply:
I was only referring to this one file - the rest of the PR is fine.

Reply:
Oh, I’ll remove that then.